pkg/steps/index_generator.go

```diff
@@ -42,15 +43,35 @@ func (*indexGeneratorStep) Validate() error { return nil }
 
 func databaseIndex(client ctrlruntimeclient.Client, name, namespace string) (bool, error) {
 	ist := &imagev1.ImageStreamTag{}
-	if err := client.Get(context.TODO(), ctrlruntimeclient.ObjectKey{Namespace: namespace, Name: name}, ist); err != nil {
+	ctx := context.TODO()
+	if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: namespace, Name: name}, ist); err != nil {
 		return false, fmt.Errorf("could not fetch source ImageStreamTag: %w", err)
 	}
+	var dockerImageMetadata runtime.RawExtension
+	var foundInManifests bool
```

Comment:
No need to use the boolean. If the image doesn't have a manifest list, we will never end up iterating there.
You can just define:
`dockerImageMetadata := image.DockerImageMetadata` and then overwrite it in the iteration.